articles/agora-express-token-server/index.md

```diff
@@ -0,0 +1,364 @@
+
+We'll generate a random user ID and set the expiration time for the token in the server.
+
+> Make sure the user ID you generate here is the same as the optional UID you pass to the SDK in the front-end while joining the channel.
+
+```JavaScript
+const appID = "<-- Your app ID here -->";
+const appCertificate = "<-- Your app certificate here -->";
+```
+
+Let's generate a random User ID. Agora accepts a 32-bit unsigned integer with a value ranging from 1 to (2^32-1) as the User ID.
+To make it simple, let's limit the range of the user ID to 100,000.
+
+```JavaScript
+const uid = Math.floor(Math.random() * 100000);
+```
+
+Let's set the expiration time of the token. You need to provide a timestamp for the expiration time. We'll add the time (in seconds) to the current timestamp to set when we want the token to expire.
+
+```JavaScript
+const expirationTimeInSeconds = 3600;
+const currentTimestamp = Math.floor(Date.now() / 1000);
+const privilegeExpiredTs = currentTimestamp + expirationTimeInSeconds;
+```
+
+There are two types of roles in a channel, a Publisher and a Subscriber.
+
+- A Publisher is someone who streams audio/video in the channel and also receives audio/video from other publishers.
+
+- A Subscriber will not be able to stream audio/video but can receive audio/video streams from publishers.
+
+For Example, In the case of a Video Conference, everyone is a publisher. But in the case of a Livestream, only some users will be publishers while the others will be subscribers.
+
+We should get whether the user is a publisher or subscriber in the request body. The `agora-access-token` provides constants for Publisher and Subscriber.
+
+```JavaScript
+const role = req.body.isPublisher ? Agora.RtcRole.PUBLISHER : Agora.RtcRole.SUBSCRIBER;
+```
+
+We should also get the Channel ID from the request body.
+
+```JavaScript
+const channel = req.body.channel;
+```
+
+Let's pass these data to `buildTokenWithUid` to generate the token.
+
+```JavaScript
+const token = Agora.RtcTokenBuilder.buildTokenWithUid(appID, appCertificate, channel, uid, role, privilegeExpiredTs);
+```
+
+Now that we have the token, Let's return the user ID and the token as the response.
+
+```JavaScript
+res.send({ uid, token });
+```
+
+Final Code:
+
+```JavaScript
+app.post("/rtctoken", (req, res) => {
+  const appID = "e004df68b67841a1b6ea7c08b7225814";
+  const appCertificate = "521c7307f1d547e080c34585dd20662e";
+  const expirationTimeInSeconds = 3600;
+  const uid = Math.floor(Math.random() * 100000);
+  const role = req.body.isPublisher ? Agora.RtcRole.PUBLISHER : Agora.RtcRole.SUBSCRIBER;
+  const channel = req.body.channel;
+  const currentTimestamp = Math.floor(Date.now() / 1000);
+  const expirationTimestamp = currentTimestamp + expirationTimeInSeconds;
+
+  const token = Agora.RtcTokenBuilder.buildTokenWithUid(appID, appCertificate, channel, uid, role, expirationTimestamp);
+  res.send({ uid, token });
+});
+```
+
+Request & Response: 
```

Comment:
Consider adding a code fence with the body of the post request.
I was also confused about what value channel represented and I used the temp token I generated on the token form.

Reply:
Oh, that's not what you're supposed to use. I'll include documentation links and explain about this.

Reply:
I've added a documentation link for `channel`. I hope that helps.

Reply:
I might be missing something
But even after reading the docs, I don't understand where to get the channel ID used to make the first POST request

Reply:
@zolomohan

Reply:
The Channel ID comes from the Request Body, which comes from the Client App.

Reply:
If you have an Agora app set up already, you need to join channels to use the RTC SDK. Think of it as a TV channel, where the video is being broadcasted, and anyone can subscribe to it and watch it.

Reply:
For example, Using Agora, we can create a Livestream, and that Livestream will be running on a channel. We need that Channel ID, to generate the auth token.

Reply:
The Channel ID can literally be any string. It can be alphanumeric.

Reply:
Right...
and that only works when you actually have a channel you want to tune into
Okay
I didn't have that bit of context
Thank you

Reply:
Exactly 😀